[PDP-1039] Add documentation for Allowed IP Addresses in cluster agent tokens (#2250)
buildkite/docs: pages/agent/v3/securing.md

## Signing pipelines

> If using plugins is crucial to your workflow and you would still like to preserve strong security guarantees, take a look at the [buildkite-signed-pipeline](https://github.com/seek-oss/buildkite-signed-pipeline) tool. This tool allows uploaded steps to be signed with a secret shared by all agents, so that plugins can run without concerns of tampering by third parties.

Reviewer: Update to https://github.com/buildkite/buildkite-signed-pipeline ?

Author: ```suggestion
If using plugins is crucial to your workflow and you would still like to preserve strong security guarantees, take a look at the [buildkite-signed-pipeline](https://github.com/buildkite/buildkite-signed-pipeline) tool. This tool allows uploaded steps to be signed with a secret shared by all agents, so that plugins can run without concerns of tampering by third parties.
```

As a result, your Buildkite agent will refuse to run anything that's not a single argumentless invocation of a script that exists locally (after the `git clone` step of the setup) unless it's explicitly allowed by you.
Since the [agent](https://github.com/buildkite/agent) is open-source, if necessary you can verify that assertion to whatever degree of certainty is required.

## Restrict agent connection by IP address

> Clusters provide a mechanism protect against the misuse of agent tokens and the hijacking of agent sessions, by restricting which IP addresses are allowed to connect using a given agent token. Setting the [_Allowed IP Addresses_ attribute](/docs/clusters/manage-clusters#set-up-clusters-restrict-access-for-a-cluster-token-by-ip-address) on a token will lock down agent registration, and forcefully disconnect any existing agents that fall outside the allowed IP ranges.

Reviewer 2: This is technically fine, but there are a couple of changes I think are worth considering:

- Splitting up the first long sentence into two smaller ones to improve readability.
- Creating a second paragraph that's more action focussed on the task.

Thinking:

```suggestion
Clusters provide a mechanism to restrict which IP addresses can connect using a given agent token. This protects against the misuse of agent tokens and the hijacking of agent sessions.

To restrict agent connection by IP address, set the [_Allowed IP Addresses_ attribute](/docs/clusters/manage-clusters#set-up-clusters-restrict-access-for-a-cluster-token-by-ip-address) with a list of the allowed IP addresses. Once set, agent registration is locked down, and any existing agents outside the allowed IP ranges are forcefully disconnected.
```

What do you think?

As a result, your Buildkite agent will refuse to run anything that's not a single argumentless invocation of a script that exists locally (after the `git clone` step of the setup) unless it's explicitly allowed by you.
Since the [agent](https://github.com/buildkite/agent) is open-source, if necessary you can verify that assertion to whatever degree of certainty is required.

## Restrict agent connection by IP address

Clusters provide a mechanism to restrict which IP addresses can connect using a given agent token. This protects against the misuse of agent tokens and the hijacking of agent sessions.

> To restrict agent connection by IP address, set the [_Allowed IP Addresses_ attribute](/docs/clusters/manage-clusters#set-up-clusters-restrict-access-for-a-cluster-token-by-ip-address) with a list of the allowed IP addresses. Once set, agent registration is locked down, and any existing agents outside the allowed IP ranges are forcefully disconnected.

Reviewer 3: Could maybe remove some words, and adjust some others…

```suggestion
To restrict agent connection by IP address, set the [_Allowed IP Addresses_ attribute](/docs/clusters/manage-clusters#set-up-clusters-restrict-access-for-a-cluster-token-by-ip-address). This restricts agent registration to those IPs, and any existing agents outside the allowed IP ranges will be forcefully disconnected.
```
